node/worker/worker.go

```diff
@@ -239,9 +243,17 @@ func (w *Worker) CommitTransactions(
 	if w.current.gasPool == nil {
 		w.current.gasPool = new(core.GasPool).AddGas(w.current.header.GasLimit())
 	}
+	erroredTxns := []types.RPCTransactionError{}
+
 	for _, tx := range txs {
 		snap := w.current.state.Snapshot()
 		_, err := w.commitTransaction(tx, coinbase)
+		if err != nil {
+			erroredTxns = append(erroredTxns, types.RPCTransactionError{
+				tx.Hash().Hex(), time.Now().Unix(), err.Error(),
+			})
+		}
+
 		if err != nil {
```

Comment:
The two checks should be combined and also `txnErrorSink(erroredTxns)` should happen inside the branch. Otherwise, the second if condition will skip it and return right?

Reply:
good catch - this file changed heavily from s3 to master so this mistake